### ISSUE TITLE
SNS Subscription appears to be incomplete

### ISSUE THREAD

**Issue by apmclean** [cloudtools/troposphere#632]
I'm working on a project using AWS::SNS::Subscription

Here's the CFN documentation for reference:
http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-sns-subscription.html#cfn-sns-topicarn

Reviewing the troposphere/sns.py it looks like Subscription is incomplete:
```
class Subscription(AWSProperty):
    props = {
        'Endpoint': (basestring, True),
        'Protocol': (basestring, True),
    }
```
I believe this should be:
```
class Subscription(AWSObject):
    resource_type = "AWS::SNS::Subscription"

    props = {
        'Endpoint': (basestring, True),
        'Protocol': (basestring, True),
        'TopicArn': (basestring, True),
    }
```

I've tested this in my CFN templates and it's worked as expected.  I can do a pull request if you prefer.  This seemed small enough that maybe someone could just sneak it in :)

**Comment by markpeek** [comment]
I've been meaning to get around to resolving this issue or, more to the point, providing the right work around for it. The initial Subscription was there in support of a [Subscription property](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-sns-subscription.html) for SNS::Topic. Recently the [Subscription Resource](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-sns-subscription.html) was added which causes a naming conflict. This has happened before (NetworkInterface and NetworkInterfaceProperty) but this time the property was around first. So now I need to decide whether to break existing code my renaming the property or pushing the naming change to the resource. @lil-cain submitted #622 as a PR for this same change to use SubscriptionObject but I've been mulling over whether it should be the more descriptive SubscriptionResource instead. This is something I mentioned to the AWS CloudFormation team at re:invent as an issue but we haven't pursued a longer term fix for these naming conflicts.

**Comment by apmclean** [comment]
Thanks Mark,

That makes more sense, I hadn’t used / realized there was an existing one with resource.

I’d be fine with SubscriptionObject.  I think with something in the ‘/examples/’ directory it would be obvious enough.  I think that’s better than not having a construct at all ☺

Thanks for the hard work on the project, I find it invaluable!  I’ll try to contribute more examples and possibly code as I do more non-trivial things.

**Comment by markpeek** [comment]
This was included in v1.9.1 via #622.
